api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubPullRequestV3.java

```diff
@@ -0,0 +1,503 @@
+
+        if ((prData == null) || prData.isEmpty()) { return "Pull Request Data Empty From "+gitHubParsed.getGraphQLUrl(); }
+
+        Object base = restClient.getAsObject(pullRequestObject, "base");
+        String branch = restClient.getString(base, "ref");
+
+        CollectorItem collectorItem = getCollectorItem(gitHubParsed.getUrl(), branch);
+
+        GitRequest pull = buildGitRequestFromPayload(repoUrl, branch, pullRequestObject, collectorItem);
+
+        updateGitRequestWithGraphQLData(pull, repoUrl, branch, prData, token);
+
+        save(pull);
+
+        return "Pull Request Processed Successfully";
+    }
+
+    protected JSONObject buildGraphQLQuery(GitHubParsed gitHubParsed, Object pullRequestObject) {
+        StringBuilder queryBuilder = new StringBuilder("");
+
+        int pullNumber = restClient.getInteger(pullRequestObject,"number");
+        int commitsCount = restClient.getInteger(pullRequestObject, "commits");
+        int commentsCount = restClient.getInteger(pullRequestObject, "comments");
+
+        if (commitsCount == 0) { return null; }
+
+        JSONObject variableJSON = new JSONObject();
+        variableJSON.put("owner", gitHubParsed.getOrgName());
+        variableJSON.put("name", gitHubParsed.getRepoName());
+        variableJSON.put("number", pullNumber);
+
+        queryBuilder.append(GraphQLQuery.PR_GRAPHQL_BEGIN_PRE);
+        if (commitsCount > 0) {
+            queryBuilder.append(GraphQLQuery.PR_GRAPHQL_COMMITS_BEGIN);
+            variableJSON.put("commits", commitsCount);
+        }
+        if (commentsCount > 0) {
+            queryBuilder.append(GraphQLQuery.PR_GRAPHQL_COMMENTS_BEGIN);
+            variableJSON.put("comments", commentsCount);
+        }
+        queryBuilder.append(GraphQLQuery.PR_GRAPHQL_BEGIN_POST);
+
+        if (commitsCount > 0) {
+            queryBuilder.append(GraphQLQuery.PR_GRAPHQL_COMMITS);
+        }
+        if (commentsCount > 0) {
+            queryBuilder.append(GraphQLQuery.PR_GRAPHQL_COMMENTS);
+        }
+
+        queryBuilder.append(GraphQLQuery.PR_GRAPHQL_REVIEWS);
+        queryBuilder.append(GraphQLQuery.PR_GRAPHQL_END);
+
+        JSONObject query = new JSONObject();
+
+        query.put("query", queryBuilder.toString());
+        query.put("variables", variableJSON.toString());
+
+        return query;
+    }
+
+    protected void save (GitRequest pull) {
+        long start = System.currentTimeMillis();
+
+        GitRequest existingPR
+        = gitRequestRepository.findByScmUrlIgnoreCaseAndScmBranchIgnoreCaseAndNumberAndRequestTypeIgnoreCase(pull.getScmUrl(), pull.getScmBranch(), pull.getNumber(), "pull");
+
+        if (existingPR != null) {
+            pull.setId(existingPR.getId());
+            pull.setCollectorItemId(existingPR.getCollectorItemId());
+        }
+
+        gitRequestRepository.save(pull);
+
+        long end = System.currentTimeMillis();
+
+        LOG.debug("Time to make gitRequestRepository call = "+(end-start));
```

Comment:
please change the verbiage to be more meaningful

Reply:
Ok, have updated this, thanks.